state/indexer/block/kv/util.go

```diff
@@ -74,18 +84,115 @@ func parseValueFromEventKey(key []byte) (string, error) {
+		height                        int64
+		eventSeq                      int64
+	)
+
+	remaining, err := orderedcode.Parse(string(key), &compositeKey, &eventValue, &height, &typ)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse event key: %w", err)
+	}
+
+	// This is done to support previous versions that did not have event sequence in their key
 	if len(remaining) != 0 {
-		return "", fmt.Errorf("unexpected remainder in key: %s", remaining)
+		remaining, err = orderedcode.Parse(remaining, &eventSeq)
+		if err != nil {
+			return 0, fmt.Errorf("failed to parse event key: %w", err)
+		}
+		if len(remaining) != 0 {
+			return 0, fmt.Errorf("unexpected remainder in key: %s", remaining)
+		}
 	}
 
-	return eventValue, nil
+	return eventSeq, nil
+}
+
+// Remove all occurrences of height equality queries except one. While we are traversing the conditions, check whether the only condition in
+// addition to match events is the height equality or height range query. At the same time, if we do have a height range condition
+// ignore the height equality condition. If a height equality exists, place the condition index in the query and the desired height
+// into the heightInfo struct
+func dedupHeight(conditions []syntax.Condition) (dedupConditions []syntax.Condition, heightInfo HeightInfo, found bool) {
+	heightInfo.heightEqIdx = -1
+	heightRangeExists := false
+	var heightCondition []syntax.Condition
+	heightInfo.onlyHeightEq = true
+	heightInfo.onlyHeightRange = true
+	for _, c := range conditions {
+		if c.Tag == types.BlockHeightKey {
+			if c.Op == syntax.TEq {
+				if found || heightRangeExists {
+					continue
+				} else {
+					heightCondition = append(heightCondition, c)
+					heightInfo.height = int64(c.Arg.Number())
+
+					found = true
+				}
+			} else {
+				heightInfo.onlyHeightEq = false
+				heightRangeExists = true
+				dedupConditions = append(dedupConditions, c)
+			}
+		} else {
+			heightInfo.onlyHeightRange = false
+			heightInfo.onlyHeightEq = false
+			dedupConditions = append(dedupConditions, c)
+		}
+	}
+	if !heightRangeExists && len(heightCondition) != 0 {
+		heightInfo.heightEqIdx = len(dedupConditions)
+		heightInfo.onlyHeightRange = false
+		dedupConditions = append(dedupConditions, heightCondition...)
+	} else {
+		// If we found a range make sure we set the hegiht idx to -1 as the height equality
+		// will be removed
+		heightInfo.heightEqIdx = -1
+		heightInfo.height = 0
+		heightInfo.onlyHeightEq = false
+		found = false
+	}
+	return dedupConditions, heightInfo, found
+}
+
+func checkHeightConditions(heightInfo HeightInfo, keyHeight int64) bool {
+	if heightInfo.heightRange.Key != "" {
+		if !checkBounds(heightInfo.heightRange, keyHeight) {
+			return false
+		}
+	} else {
+		if heightInfo.height != 0 && keyHeight != heightInfo.height {
+			return false
+		}
+	}
+	return true
 }
 
+//nolint:unused,deadcode
 func lookForHeight(conditions []syntax.Condition) (int64, bool) {
```

Comment:
I think we should delete this function here entirely, because it is not used. We left it in old versions to not delete functions people may be using. 
I tried to repurpose this function initially (hence the 3rd value), then we decided not to use it and wrote the `dedupHeight` function because we needed to remove multiple occurrences of height in the query when searching with `match_events=true`. However, in 0.34 the 3rd return value was needed because we allowed for the old behaviour. `0.37.x` and `main` are never calling this function.

Reply:
I see, so you applied a small refactoring in this PR to get rid of the unneeded value?

Reply:
Yes, but I think we should delete the function entirely for this PR. Honestly I think this should have been done for 0.37 but it was forgotten. Do you think it would break anything if we were to delete it?

Reply:
I removed the functions, they are not used, this is a major release so no need to keep them around.